### PR TITLE
Add comprehensive test coverage and version validation

### PR DESCRIPTION
**Pull request by mattgle**
### Summary

- **New test file**: `test/chain.test.ts` with 23 tests covering all chain utility functions (`xorRailgun`, `getChainFullNetworkID`, `networkIDToChain`, `chainToNetworkID`, `is0zk`)
- **Enhanced tests**: Added 7 codec tests for edge cases (version validation, boundary chain IDs, different keys, common EVM chains)
- **Version validation**: Added validation in `parse()` and `stringify()` to reject unsupported address versions
- **Type improvements**: Replaced `Optional<T>` with explicit `T | undefined`

### Test Coverage

- ✅ Chain utilities: XOR reversibility, boundary values (0, max chain ID), common chains (Polygon, Arbitrum, BSC)
- ✅ Address validation: Invalid prefixes, malformed addresses, key length validation
- ✅ Edge cases: Zero chain ID, max chain ID, different master/viewing keys, all-0xFF keys

### CODE REVIEW

File: test/chain.test.ts [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+    it('Should return encoded network ID for valid chain', () => {
+      const chain: Chain = { type: ChainType.EVM, id: 1n }
+      const result = chainToNetworkID(chain)
+
+      assert.strictEqual(result.length, 8)
+      assert.strictEqual(result[0], ChainType.EVM)
+    })
+
+    it('Should return ALL_CHAINS_NETWORK_ID when chain is null', () => {
+      const result = chainToNetworkID(undefined)
```
Comment (bhflm) [inline test/chain.test.ts]: this is the very same test as line 193, its duplicated ? or pass null :p

File: test/chain.test.ts [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+
+        const dataView = new DataView(networkID.buffer)
+        const fullValue = dataView.getBigUint64(0, false)
+        assert.strictEqual(fullValue & CHAIN_ID_MASK, id, `Failed for ${name}`)
+      }
+    })
+  })
+
+  describe('networkIDToChain', () => {
+    it('Should decode EVM chain ID 1 correctly', () => {
```
Comment (bhflm) [inline test/chain.test.ts:166]: can't we use something like `getChainFullNetworkID` to construct test data ? 

ie: 

```
const chain: Chain = { type: ChainType.EVM, id: 1n }
const encoded = getChainFullNetworkID(chain)
... 

```


then do smth like `decoded = networkIdToChain` etc etc ??

File: test/chain.test.ts
Comment (bhflm) [inline test/chain.test.ts:1]: i'd add one more test, what happens if you exceed chain_id_mask ?? (more than 56 bits, try passing 2n**60n to `getChainFullNetworkID`)

**Review by bhflm — CHANGES_REQUESTED** [review]

File: test/chain.test.ts (hunk 1 above)
Comment (mattgle) [reply test/chain.test.ts]: Yeah ill delete it, good catch

File: test/chain.test.ts (hunk 2 above)
Comment (mattgle) [reply test/chain.test.ts:166]: absolutely

**Review by bhflm — APPROVED** [review]
